src/consumerIncentives/saga.ts

```diff
@@ -126,18 +150,78 @@ function* claimReward(reward: SuperchargePendingReward, index: number, baseNonce
   }
 }
 
+function* claimRewardV2(reward: SuperchargePendingRewardV2, index: number, baseNonce: number) {
+  const { transaction, details } = reward
+
+  const superchargeRewardContractAddress = yield select(superchargeRewardContractAddressSelector)
+  if (superchargeRewardContractAddress !== transaction.to) {
```

Comment:
Remember that in the V2 endpoint we will also return the pending rewards from the previous program. So the `transation.to` could be different in those cases.

Reply:
ahhhh such a good catch, thanks! damn, i think i'll need to add another remote config for this 🥲

Reply:
aaaaah damn. the `to` address is from the merkle contract 😵‍💫 i need to think about this some more

Reply:
added this! https://github.com/valora-inc/wallet/pull/3475